Lib/urllib/parse.py

```diff
@@ -504,9 +514,6 @@ def _urlsplit(url, scheme=None, allow_fragments=True):
         if (('[' in netloc and ']' not in netloc) or
                 (']' in netloc and '[' not in netloc)):
             raise ValueError("Invalid IPv6 URL")
-        if '[' in netloc and ']' in netloc:
```

Comment:
This should be kept and updated to error appropriately as callers of urlsplit and urlparse expect the exception at parsing time.  Not later on when accessing attributes.